compiler-core/src/codegen.rs

```diff
@@ -242,8 +257,30 @@ impl<'a> JavaScript<'a> {
             &module.code,
             self.target_support,
             self.typescript,
+            &mut source_map_emitter,
         );
         tracing::debug!(name = ?js_name, "Generated js module");
-        writer.write(&path, &output?)
+        writer.write(&path, &output?)?;
+
+        match source_map_emitter {
+            SourceMapEmitter::Null => Ok(()),
+            SourceMapEmitter::Emit(builder) => {
+                let sourcemap_name = format!("{js_name}.mjs.map");
+                let sourcemap_path = self.output_directory.join(sourcemap_name);
+                tracing::debug!(path = ?sourcemap_path, name = ?js_name, "Emitting sourcemap for module");
+
+                // NOTE: This is a bit inefficient
+                // * we first write to a buffer
+                // * then construct a String based on the output
+                // * then write to the output
+                let sourcemap = builder.into_sourcemap();
+
+                let mut output = Vec::new();
+                sourcemap.to_writer(&mut output).expect("Failed to write sourcemap to memory. This is a bug in sourcemap, please report.");
+                let output = String::from_utf8(output).expect("Sourcemap did not generate valid UTF-8. This is a bug in sourcemap, please report.");
```

Comment:
Really not a fan of those `expect` - opened to any suggestion here

Reply:
Please remove the bit about opening a bug ticket, all panics have this information printed for them